src/table/types.ts

```diff
@@ -0,0 +1,218 @@
+  cursor?: string;
+  borderWidth?: string;
+  backgroundColor?: string;
+  sortLabelColor?: string;
+  hoverBackgroundColor?: string;
+  hoverFontColor?: string;
+}
+
+export interface CellStyle {
+  backgroundColor: string | undefined; // This is always set but could be undefined in the theme
+  color: string;
+  background?: string;
+  selectedCellClass?: string;
+}
+
+export interface HandleWrapperKeyDownProps {
+  evt: React.KeyboardEvent;
+  totalRowCount: number;
+  page: number;
+  rowsPerPage: number;
+  handleChangePage(pageIdx: number): void;
+  setShouldRefocus(): void;
+  keyboard: stardust.Keyboard;
+  isSelectionMode: boolean;
+}
+
+export interface HandleHeadKeyDownProps {
+  evt: React.KeyboardEvent;
+  rootElement: HTMLElement;
+  cellCoord: [number, number];
+  column: Column;
+  changeSortOrder: ChangeSortOrder;
+  layout: TableLayout;
+  isSortingEnabled: boolean;
+  setFocusedCellCoord: React.Dispatch<React.SetStateAction<[number, number]>>;
+}
+
+export interface HandleBodyKeyDownProps {
+  evt: React.KeyboardEvent;
+  rootElement: HTMLElement;
+  cell: Cell;
+  selectionDispatch: React.Dispatch<TSelectionActions>;
+  isSelectionsEnabled: boolean;
+  setFocusedCellCoord: React.Dispatch<React.SetStateAction<[number, number]>>;
+  announce: Announce;
+  keyboard: stardust.Keyboard;
+  totalsPosition: TotalsPosition;
+  paginationNeeded: boolean;
+  selectionsAPI: ExtendedSelectionAPI;
+}
+
+export interface CellFocusProps {
+  focusType: string;
+  cell: HTMLTableCellElement | undefined;
+}
+
+export interface HandleResetFocusProps {
+  focusedCellCoord: [number, number];
+  rootElement: HTMLElement;
+  shouldRefocus: React.MutableRefObject<boolean>;
+  isSelectionMode: boolean;
+  setFocusedCellCoord: React.Dispatch<React.SetStateAction<[number, number]>>;
+  keyboard: stardust.Keyboard;
+  announce: Announce;
+}
+
+export interface ContextProviderProps {
+  children: JSX.Element;
+  selectionsAPI: ExtendedSelectionAPI;
+  cellCoordMock?: [number, number];
+  selectionDispatchMock?: jest.Mock<any, any>;
+}
+
+export interface RenderProps {
+  direction?: 'ltr' | 'rtl';
+  selectionsAPI: ExtendedSelectionAPI;
+  rootElement?: HTMLElement;
+  layout: TableLayout;
+  changeSortOrder: ChangeSortOrder;
+  rect: stardust.Rect;
+  tableData?: TableData;
+  pageInfo?: PageInfo;
+  setPageInfo?: SetPageInfo;
+  constraints?: stardust.Constraints;
+  translator?: ExtendedTranslator;
+  theme: ExtendedTheme;
+  keyboard?: stardust.Keyboard;
+  footerContainer?: HTMLElement;
+  announce?: Announce;
+  model?: EngineAPI.IGenericObject;
+  manageData?(
+    model: EngineAPI.IGenericObject | undefined,
+    layout: TableLayout,
+    pageInfo: PageInfo,
+    setPageInfo: SetPageInfo
+  ): Promise<TableData | null>;
+  app?: EngineAPI.IApp;
+}
+
+export interface CommonTableProps {
+  tableData: TableData;
+  theme: ExtendedTheme;
+  keyboard: stardust.Keyboard;
+}
+
+export interface TableWrapperProps extends CommonTableProps {
```

Comment:
Should this one and the next two extend from `TableTotalProps` aswell since they all use rootElement and layout?

Reply:
I thought about it, but let's say we add a prop to totals that the other three don't need, then we would have to revert it to the way it is now. We will be adding a lot features soon so these props are likely to change

Reply:
Makes sense, then leave it as it is until you know more.